### PR TITLE
Support decoding valid json containing lists of type `List<dynamic>`

### PR DESCRIPTION
**Pull request by nshoura**
**Closes: #31**

Attempt to cast `List<dynamic>` to `List<T>` when parsing an RRule set.

### Checklist

- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: lib/src/codecs/json/decoder.dart [hunk 1]
```diff
@@ -70,6 +70,12 @@ class RecurrenceRuleFromJsonDecoder
       return {parse(json)};
     } else if (json is List<T>) {
       return json.map(parse).toSet();
+    } else if (json is List) {
+      try {
+        return json.cast<T>().map(parse).toSet();
+      } catch(_) {
+        throw FormatException('Invalid JSON in `$name`.');
```
Comment (JonasWanke) [inline lib/src/codecs/json/decoder.dart:77]:
```suggestion
      } catch(e, st) {
        throw FormatException('Invalid JSON in `$name`: $e\n\n$st');
```

File: test/codecs/json/json_test.dart [hunk 2]
```diff
@@ -25,6 +25,25 @@ void main() {
...
+        ByWeekDayEntry(DateTime.sunday, -1),
+        ByWeekDayEntry(DateTime.monday, 2),
+      },
+      byMonths: {10},
+    ),
+    json: <String, dynamic>{
+      'freq': 'YEARLY',
+      'count': 5,
+      'byday': <dynamic>['-1SU', '2MO'],
+      'bymonth': [10],
```
Comment (JonasWanke) [inline test/codecs/json/json_test.dart:43]:
```suggestion
      'bymonth': <dynamic>[10],
```

**Review by JonasWanke — CHANGES_REQUESTED** [review]
Thanks for the clear issue description and for creating a PR! I've only added two minor comments before this can get merged.

**Comment by nshoura** [comment]
Thanks for the input @JonasWanke  :) Committed the proposed changes.

**Review by JonasWanke — APPROVED** [review]

**Comment by JonasWanke** [comment]
Thanks for the quick changes! This PR is now published as part of [v0.2.8](https://pub.dev/packages/rrule/changelog#028--2022-05-05). Thanks for contributing!

**Pull request merged**